build-tools/packages/version-tools/src/semver.ts

```diff
@@ -187,3 +189,74 @@ export function isPrereleaseVersion(version: string | semver.SemVer | undefined)
 
     return prerelease !== null && prerelease.length > 0;
 }
+
+/**
+ * Calculates and returns the previous major and minor versions for the provided version, taking into account the Fluid
+ * internal and virtualPatch version schemes.
+ *
+ * This function does not consult any external sources to determine what versions are available. In other words, it
```

Comment:
Nit: this paragraph seems suited for a `@remarks` block, rather than being in the summary IMO.